pkg/collector/corechecks/net/network/network.go

```diff
@@ -130,58 +153,72 @@ func (c *NetworkCheck) Run() error {
 	if err != nil {
 		return err
 	}
-	for _, interfaceIO := range ioByInterface {
-		if !c.isDeviceExcluded(interfaceIO.Name) {
-			submitInterfaceMetrics(sender, interfaceIO)
-		}
-	}
 
-	protocols := []string{"tcp", "udp"}
-	protocolsStats, err := c.net.ProtoCounters(protocols)
+	protocols := []string{"Tcp", "TcpExt", "Ip", "IpExt", "Udp"}
+	counters, err := c.net.NetstatAndSnmpCounters(protocols)
 	if err != nil {
-		return err
+		log.Debug(err)
+	} else {
+		for _, protocol := range protocols {
+			if _, ok := counters[protocol]; ok {
+				submitProtocolMetrics(sender, counters[protocol])
+			}
+		}
 	}
-	for _, protocolStats := range protocolsStats {
-		// For TCP we want some extra counters coming from /proc/net/netstat if available
-		if protocolStats.Protocol == "tcp" {
-			counters, err := c.net.NetstatTCPExtCounters()
-			if err != nil {
-				log.Debug(err)
-			} else {
-				for counter, value := range counters {
-					protocolStats.Stats[counter] = value
+
+	for _, interfaceIO := range ioByInterface {
+		if !c.isDeviceExcluded(interfaceIO.Name) {
+			submitInterfaceMetrics(sender, interfaceIO)
+			if c.config.instance.CollectEthtoolStats {
+				err = handleEthtoolStats(sender, interfaceIO)
+				if err != nil {
+					return err
 				}
 			}
 		}
-		submitProtocolMetrics(sender, protocolStats)
 	}
 
 	if c.config.instance.CollectConnectionState {
+		ssAvailable := false
+		if ssAvailableFunction() == nil {
+			ssAvailable = true
+		}
 		connectionsStats, err := c.net.Connections("udp4")
 		if err != nil {
 			return err
 		}
-		submitConnectionsMetrics(sender, "udp4", udpStateMetricsSuffixMapping, connectionsStats)
+		submitConnectionsMetrics(sender, "udp4", udpStateMetricsSuffixMapping, connectionsStats, c.config.instance.CollectConnectionQueues, ssAvailable)
```

Comment:
Suggestion: Since we only collect connection queue information for the tcp protocol, what about setting false directly on the udp calls directly? 

```
submitConnectionsMetrics(sender, "udp4", udpStateMetricsSuffixMapping, connectionsStats, false, ssAvailable)
```

That way in the `submitConnectionsMetrics` we do not have to check the protocol name:

We would change `if collectConnectionQueues && protocolName[:3] == "tcp" {` to just `if collectConnectionQueues {`

Reply:
cleaned this up in https://github.com/DataDog/datadog-agent/pull/33190/commits/6cb3260b4969492040c7754bd997f740c8e99247